crates/ruff/src/rules/pycodestyle/rules/lambda_assignment.rs

```diff
@@ -2,64 +2,88 @@ use ruff_macros::{define_violation, derive_message_formats};
 use rustpython_parser::ast::{Arguments, Expr, ExprKind, Location, Stmt, StmtKind};
 
 use crate::ast::helpers::{match_leading_content, match_trailing_content, unparse_stmt};
-use crate::ast::types::Range;
+use crate::ast::types::{Range, ScopeKind};
 use crate::ast::whitespace::leading_space;
 use crate::checkers::ast::Checker;
 use crate::fix::Fix;
 use crate::registry::Diagnostic;
 use crate::source_code::Stylist;
-use crate::violation::AlwaysAutofixableViolation;
+use crate::violation::{AutofixKind, Availability, Violation};
 
 define_violation!(
-    pub struct LambdaAssignment(pub String);
+    pub struct LambdaAssignment {
+        pub name: String,
+        pub fixable: bool,
+    }
 );
-impl AlwaysAutofixableViolation for LambdaAssignment {
+impl Violation for LambdaAssignment {
+    const AUTOFIX: Option<AutofixKind> = Some(AutofixKind::new(Availability::Sometimes));
+
     #[derive_message_formats]
     fn message(&self) -> String {
         format!("Do not assign a `lambda` expression, use a `def`")
     }
 
-    fn autofix_title(&self) -> String {
-        let LambdaAssignment(name) = self;
-        format!("Rewrite `{name}` as a `def`")
+    fn autofix_title_formatter(&self) -> Option<fn(&Self) -> String> {
+        let LambdaAssignment { fixable, .. } = self;
+        if *fixable {
+            Some(|v| format!("Rewrite `{}` as a `def`", v.name))
+        } else {
+            None
+        }
```

Comment:
maybe you like this form
```suggestion
        self.fixable
            .then_some(|v| format!("Rewrite `{}` as a `def`", v.name))
```

Reply:
I do!